bin/docker.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# Author: Valentin Kuznetsov <vkuznet@gmail.com>
+# helper script to handle docker actions
+
+if [ $# -ne 4 ]; then
+     echo "Usage: docker.sh <action> <service> <tag> <registry> "
+     echo "Supported actions: build, build-stable, push"
+     exit 1;
+fi
+
+# helper function to match the pattern X.Y.Z where X, Y, and Z are numbers
+function match_tag {
+  local tag=$1
+  if [[ $tag =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+    return 0
+  elif [[ $tag =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+# make input variable assignments
+action=$1
+service=$2
+tag=$3
+registry=$4
+rurl=$registry/cmsweb/$service
+
+# check if action is supported
+case "$action" in
+  "build"|"build-stable"|"push"|"remove"|"delete")
+    ;;
+  *)
+    echo "action: '$action' is not supported"
+    exit 1
+    ;;
+esac
+
+# determine suffix of image tag
+suffix=""
+if match_tag "$tag"; then
+    suffix="-stable"
+fi
+
+# check build/push actions and build the image(s) if it will be required
+if [ "$action" == "build" ]; then
+    echo "check if image with service=$service and tag=$tag exist ..."
+    image_exist=`docker images | grep $tag | grep $service`
+    if [ -z "$image_exist" ]; then
+        echo "action: docker build --build-arg TAG=${tag} --tag ${rurl}:${tag}"
+        docker build --build-arg TAG=${tag} --tag ${rurl}:${tag} .
+    fi
+    docker images | grep $tag | grep $service
+fi
```

Comment:
I would suggest removing this `fi` line and refactor the `if` statement below to something as simple as:
```
if [ -n "$suffix" ]; then
```
properly indented under the `build` action.